src/sagas/session.js

```diff
@@ -24,16 +24,30 @@ export function* getServerInfo(
   action: ActionType<typeof actions.getServerInfo>
 ): SagaGen {
   const { auth } = action;
+
+  setupClient(auth);
+  // Fetch server information
+  const client = getClient();
   try {
-    setupClient(auth);
-    // Fetch server information
-    const client = getClient();
     const serverInfo = yield call([client, client.fetchServerInfo]);
+
+    // Check that the client was not changed in the mean time.
```

Comment:
Maybe add something a bit more explicative? Something like:
```
// Check that the client was not changed in the mean time. This could happen if a request to
// another server was sent after the current one, but took less time to answer.
// In such a case, this means this saga/server request should be discarded in favor of the other one
// which was sent later.
```